HIP/hip-1056.md

```diff
@@ -0,0 +1,1822 @@
+        proto.NetworkStakingRewards network_staking_rewards = 5;
+
+        /**
+        * A change to a raw byte array singleton.
+        */
+        google.protobuf.BytesValue bytes = 6;
+
+        /**
+        * A change to a raw string singleton.
+        */
+        google.protobuf.StringValue string = 7;
+
+        /**
+        * A change to the running hashes singleton.
+        */
+        proto.RunningHashes running_hashes = 8;
+
+        /**
+        * A change to the throttle usage snapshots singleton.
+        * <p>
+        * Throttle usage snapshots SHALL be updated for _every transaction_
+        * to reflect the amount used for each tps throttle and
+        * for the gas throttle.
+        */
+        proto.ThrottleUsageSnapshots throttle_usage_snapshots = 9;
+
+        /**
+        * A change to a raw `Timestamp` singleton.<br/>
+        * An example of a raw `Timestamp` singleton is the
+        * "network freeze time" singleton state, which, if set, stores
+        * the time for the next scheduled freeze.
+        */
+        proto.Timestamp timestamp = 10;
+
+        /**
+        * A change to the block stream status singleton.
+        */
+        com.hedera.hapi.node.state.blockstream.BlockStreamInfo block_stream_info = 11;
+
+        /**
+        * A change to the platform state singleton.
+        */
+        com.hedera.hapi.platform.state.PlatformState platform_state = 12;
+
+        /**
+        * A change to the roster state singleton. 
+        */
+        com.hedera.hapi.node.state.roster.RosterState roster_state = 13;
+    }
+}
+```
+
+```protobuf
+/**
+ * An update to a single item in a `VirtualMap`.<br/>
+ * Each update consists of a "key" and a "value".
+ * Keys are often identifiers or scalar values.
+ * Values are generally full messages or byte arrays.
+ */
+message MapUpdateChange {
+    /**
+    * A key in a virtual map.
+    * <p>
+    * This key MUST be mapped to the value added or updated.<br/>
+    * This field is REQUIRED.
+    */
+    MapChangeKey key = 1;
+
+    /**
+    * A value in a virtual map.
+    * <p>
+    * This value MUST correctly represent the state of the map entry
+    * _after_ the asserted update.<br/>
+    * This value MAY be reduced to only transmit fields that differ
+    * from the prior state.<br/>
+    * This field is REQUIRED.
+    */
+    MapChangeValue value = 2;
+}
+```
+
+```protobuf
+/**
+ * A removal of a single item from a `VirtualMap`.
+ */
+message MapDeleteChange {
+    /**
+    * A key in a virtual map.
+    * <p>
+    * This key SHALL be removed.<br/>
+    * The mapped value SHALL also be removed.<br/>
+    * This field is REQUIRED.
+    */
+    MapChangeKey key = 1;
+}
+```
+
+```protobuf
+/**
+ * A key identifying a specific entry in a key-value "virtual map".
+ */
+message MapChangeKey {
+    oneof key_choice {
+        /**
+        * A key for a change affecting a map keyed by an Account identifier.
+        */
+        proto.AccountID account_id = 1;
```

Comment:
The changes to the keys and value names were not also made to the protos in this PR. Will we be doing a mass sync from the md to the proto before merge? Or how else are we ensuring things are not getting out of sync?

Reply:
We need to resync things, definitely.  There are some substantial changes being worked on that, I suspect, will require a good bit of changes across all four locations.  I think we'll need to resync at that point.